Pass GBError to refresh handler

vazarkevych opened the pull request:
This PR adds GbError support to the RefreshHandler, allowing refresh issues to be tracked and reported through analytics.

madhuchavva commented on lib/src/MultiUserMode/constant.dart:
```diff
@@ -4,4 +4,4 @@ typedef TrackingCallBackWithUser = void Function(GBTrackData)?;
 typedef FeatureUsageCallbackWithUser = void Function(String, GBFeatureResult);
 
-typedef FeatureRefreshCallback = void Function(bool);
+typedef FeatureRefreshCallback = void Function(bool, GBError?);
```
This is a breaking change. Let us do it with backwards compatibility

madhuchavva commented on lib/src/growth_book_sdk.dart:
```diff
@@ -79,11 +82,17 @@ class GBSDKBuilderApp {
     return this;
   }
 
+  GBSDKBuilderApp setRefreshHandlerV2(CacheRefreshHandlerV2 refreshHandlerV2) {
```
`setRefreshHandlerV2(...)` stores the V2 callback, but `initialize()` never passes `refreshHandlerV2` into `GrowthBookSDK._`; only the deprecated `refreshHandler` is wired. That means users who call this new method will not receive the error-aware callback after initialization. Please pass `refreshHandlerV2` through during initialization and add a builder-level test that receives the `GBError` on refresh failure.

madhuchavva reviewed (changes requested):
Requesting changes because the new V2 refresh handler is not wired through initialization, so the functional addition is incomplete.

vazarkevych committed:
feat: add error-aware refresh handler (CacheRefreshHandlerV2)

Introduce a new refresh handler signature that receives the GBError
that caused a failure, so consumers can distinguish network errors
from parse/decryption errors.

- Add CacheRefreshHandlerV2 typedef: void Function(bool, GBError?)
- Add setRefreshHandlerV2 builder method
- Wire refreshHandlerV2 through GrowthBookSDK._() initialization
- Fire V2 handler wherever V1 was fired (features success/fail/not-modified,
  saved groups success/fail), passing null on success and the actual
  GBError on failure

Legacy handlers preserved for backwards compatibility:
- CacheRefreshHandler and GBCacheRefreshHandler typedefs marked @Deprecated
- setRefreshHandler builder method marked @Deprecated
- Both V1 and V2 handlers fire in parallel when both are registered

Includes builder-level tests verifying that V2 handler receives:
- (false, GBError) on remote refresh failure
- (true, null) on successful refresh

vazarkevych committed:
Merge remote-tracking branch 'origin/main' into feature/refresh-handler-error

# Conflicts:
#	lib/src/growth_book_sdk.dart

madhuchavva reviewed (approved):
Approved: the error-aware refresh handler is wired through the builder, preserves the deprecated V1 callback, and passes the current full test suite and analyzer.
